### PR TITLE
The Hazmat Joe

### PR DESCRIPTION
**Pull request by QuickLode**
# About the pull request
This pull request resprites the entire Working Joe from toes to head. It also gives two additional uniforms which are meant for hazardous use, and this PR should act as a foundation for future implementation of the Hazmat Joe into CM's gameplay. Additionally, I may just set this to draft and let it be reviewed while I work on the actual implementation.

They are complete with distinctive loadouts, which focus more on hazardous situations, repair, and firefighting. Though may tweak things depending on how its implemented.

# Explain why it's good for the game
This adds a fan favorite variation of your inexpensive, reliable friend! You've seen him in quite a few places, and now he's coming to CM!

Also, the resprite of the Joe fixes up some minor sprite issues that were encountered on previous models.

More content, more roleplay possibilities! ARES! Get me some Joes to put that reactor fire out ASAP!
# Testing Photographs and Procedure
https://cdn.discordapp.com/attachments/490668342357786645/1104748917398175795/image.png
https://media.discordapp.net/attachments/490668342357786645/1105643891107049572/image.png
Ran several tests and they went well.

# Changelog
:cl:QuickLoad,Frans_Feiffer,nauticall
add: Adds The Hazmat Joe with two minor variations. This is a Working Joe equipped to handle hazardous situations, dangerous repairs and firefighting! They are complete with their own gear, tasks, job and purpose. Forget the trashbag, get that wall fixed before we get spaced!
imageadd: Adds a new Working Joe model made by Frans Feiffer!
imageadd: Adds two variations of the Working Joe, aka the Hazmat Joe. Complete with accessories! Beautiful sprites by Frans Feiffer!
add: Android Maintenance Stations / Synthetic Repair Stations will remove shrapnel & fix organ damage. Working Joes no longer have knives, and should report to the stations for repair. Gigantic thanks to nauticall for her work on this!!
imagedel: Removes(replaces) the old Working Joe model.
add: Working Joes receive some basic equipment, and are slightly resilient to disarms.
add: Working Joes will start at 3, with a maximum of 6 depending on population.
add: Joes can access a Synthetic vendor to replace their uniform if it is damaged.
fix: Minor changes to PO Uniform.
/:cl:

### CODE REVIEW

File: code/modules/gear_presets/synths.dm [hunk 1]
```diff
@@ -491,6 +491,59 @@
...
+	faction = FACTION_MARINE
+	faction_group = list(FACTION_MARINE)
+	assignment = JOB_WORKING_JOE
+	rank = JOB_WORKING_JOE
+	skills = /datum/skills/working_joe
+	languages = list(LANGUAGE_ENGLISH, LANGUAGE_APOLLO, LANGUAGE_RUSSIAN, LANGUAGE_JAPANESE, LANGUAGE_GERMAN, LANGUAGE_SPANISH, LANGUAGE_CHINESE)
+
+/datum/equipment_preset/synth/working_joe/New()
+	. = ..()
+	access = get_all_accesses()
```
Comment (harryob) [inline code/modules/gear_presets/synths.dm:507]: you're calling parent here, any /datum/equipment_preset/synth/New() already gets all access, so this is redundant

**Review by harryob — APPROVED** [review]
otherwise looks good - there's the nitpick of the arguments being mob/living/carbon/human/H but that's just because you're using the inherited ones, so no big deal

**Comment by Diegoflores31** [comment]
image
this should probably go into a different PR . :atom:

File: code/modules/clothing/gloves/marine_gloves.dm [hunk 2]
```diff
@@ -163,3 +163,19 @@
...
 	name = "custom shooting gloves"
 	desc = "Highly protective against injury, temperature, and electric shock. Cool in the summer, warm in the winter, and a secure grip on any surface. You could buy a lot for the price of these, and they're worth every penny."
+
+/obj/item/clothing/gloves/marine/joe
+	name = "Seegson hazardous gloves"
+	desc = "Special Synthetic gloves made for touching and interacting with extremely hazardous materials. Resistant to biohazard liquids, corrosive materials and more. SEEGSON is proudly displayed on the back, along with a biohazard symbol. Tomorrow, Together."
+	icon_state = "working_joe"
+	item_state = "working_joe"
+	siemens_coefficient = 0
+	flags_inventory = CANTSTRIP
```
Comment (Diegoflores31) [inline code/modules/clothing/gloves/marine_gloves.dm:173]: Dont do CANTSTRIP on limbs that can be delimbed

**Review by harryob — CHANGES_REQUESTED** [review]
job options stuff needs changing post-#3354

File: code/game/jobs/job/civilians/support/working_joe.dm [hunk 3]
```diff
@@ -1,17 +1,37 @@
+#define STANDARD_VARIANT "Working Joe"
+#define HAZMAT_VARIANT "Hazmat Joe"
+
 /datum/job/civilian/working_joe
 	title = JOB_WORKING_JOE
-	total_positions = 3
+	total_positions = 6
 	spawn_positions = 3
```
Comment (BeagleGaming1) [inline code/game/jobs/job/civilians/support/working_joe.dm:7]: This should be the same as `total_positions`

File: code/game/jobs/job/civilians/support/working_joe.dm [hunk 4]
```diff
@@ -1,17 +1,37 @@
...
 	allow_additional = 1
+	scaled = 1
```
Comment (BeagleGaming1) [inline code/game/jobs/job/civilians/support/working_joe.dm:9]: You need to add the scaling formula and proc

File: code/game/jobs/job/civilians/support/working_joe.dm [hunk 5]
```diff
@@ -1,17 +1,37 @@
...
+
+/datum/job/civilian/working_joe/handle_job_options(option)
+	if(option != HAZMAT_VARIANT)
+		standard = TRUE
+		gear_preset = /datum/equipment_preset/synth/working_joe
+	else
+		standard = FALSE
+		gear_preset = /datum/equipment_preset/synth/working_joe/engi
+
+/datum/job/civilian/working_joe/generate_entry_message(mob/living/carbon/human/H)
```
Comment (BeagleGaming1) [inline code/game/jobs/job/civilians/support/working_joe.dm:26]: You could add wiki links to these entry messages

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

File: code/modules/gear_presets/synths.dm [hunk 6]
```diff
@@ -486,11 +486,60 @@
...
+
+/datum/equipment_preset/synth/working_joe/engi
+	name = "Synthetic - Hazmat Joe"
+	flags = EQUIPMENT_PRESET_START_OF_ROUND|EQUIPMENT_PRESET_MARINE
+	faction = FACTION_MARINE
+	faction_group = list(FACTION_MARINE)
+	assignment = JOB_WORKING_JOE
+	rank = JOB_WORKING_JOE
+	skills = /datum/skills/working_joe
+	languages = list(LANGUAGE_ENGLISH, LANGUAGE_APOLLO, LANGUAGE_RUSSIAN, LANGUAGE_JAPANESE, LANGUAGE_GERMAN, LANGUAGE_SPANISH, LANGUAGE_CHINESE)
```
Comment (BeagleGaming1) [inline code/modules/gear_presets/synths.dm:509]: Isn't most of this inherited from `/datum/equipment_preset/synth/working_joe`?

File: code/__DEFINES/job.dm [hunk 7]
```diff
@@ -64,6 +64,7 @@ var/global/list/job_squad_roles = JOB_SQUAD_ROLES_LIST
 #define JOB_MESS_SERGEANT "Mess Technician"
 #define JOB_SYNTH "Synthetic"
 #define JOB_WORKING_JOE "Working Joe"
+#define JOB_HAZMAT_JOE "Hazmat Joe"
```
Comment (BeagleGaming1) [inline code/__DEFINES/job.dm:67]: this define is unused

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

File: code/game/jobs/job/civilians/support/working_joe.dm [hunk 8]
```diff
@@ -1,17 +1,52 @@
...
+			positions = total_positions_so_far
+		else
+			total_positions_so_far = positions
+	else
+		total_positions_so_far = positions
+	return positions
+
+/datum/job/civilian/working_joe/generate_entry_message(mob/living/carbon/human/H)
+	if(standard)
+		. = {"You are a <a href='"+URL_WIKI_WJ_GUIDE+"'> Working Joe.</a> You are held to a higher standard and are required to obey not only the Server Rules but Marine Law, Roleplay Expectations and Synthetic Rules.  Your primary task is to maintain the cleanliness of the ship, putting things in their proper place. Alternatively, your primary task may be to assist with manual labor in limited capacity, or clerical duties. Your capacities are limited, but you have all the equipment you need, and the central AI has a plan! Stay in character at all times. Use the APOLLO link to communicate with your uplink!"}
```
Comment (BeagleGaming1) [inline code/game/jobs/job/civilians/support/working_joe.dm:43]:
```suggestion
		. = {"You are a <a href='"+URL_WIKI_WJ_GUIDE+"'>Working Joe.</a> You are held to a higher standard and are required to obey not only the Server Rules but Marine Law, Roleplay Expectations and Synthetic Rules.  Your primary task is to maintain the cleanliness of the ship, putting things in their proper place. Alternatively, your primary task may be to assist with manual labor in limited capacity, or clerical duties. Your capacities are limited, but you have all the equipment you need, and the central AI has a plan! Stay in character at all times. Use the APOLLO link to communicate with your uplink!"}
```

File: code/game/jobs/job/civilians/support/working_joe.dm [hunk 9]
```diff
@@ -1,17 +1,52 @@
...
+	else
+		. = {"You are a <a href='"+URL_WIKI_WJ_GUIDE+"'> Working Joe</a> for Hazardous Environments!  You are held to a higher standard and are required to obey not only the Server Rules but Marine Law, Roleplay Expectations and Synthetic Rules.  You are a variant of the Working Joe built for tougher environments and fulfill the specific duty of dangerous repairs or maintenance. Your primary task is to maintain the reactor, SMES and AI Core. Your secondary task is to respond to hazardous environments, such as an atmospheric breach or biohazard spill, and assist with repairs when ordered to by either an AI Mainframe, or a Commisioned Officer.  You should not be seen outside of emergencies besides in Engineering and the AI Core! Stay in character at all times. Use the APOLLO link to communicate with your uplink!"}
```
Comment (BeagleGaming1) [inline code/game/jobs/job/civilians/support/working_joe.dm:45]:
```suggestion
		. = {"You are a <a href='"+URL_WIKI_WJ_GUIDE+"'>Working Joe</a> for Hazardous Environments!  You are held to a higher standard and are required to obey not only the Server Rules but Marine Law, Roleplay Expectations and Synthetic Rules.  You are a variant of the Working Joe built for tougher environments and fulfill the specific duty of dangerous repairs or maintenance. Your primary task is to maintain the reactor, SMES and AI Core. Your secondary task is to respond to hazardous environments, such as an atmospheric breach or biohazard spill, and assist with repairs when ordered to by either an AI Mainframe, or a Commisioned Officer.  You should not be seen outside of emergencies besides in Engineering and the AI Core! Stay in character at all times. Use the APOLLO link to communicate with your uplink!"}
```

File: code/game/jobs/job/civilians/support/working_joe.dm [hunk 10]
```diff
@@ -1,17 +1,52 @@
...
-	allow_additional = 1
-	supervisors = "ARES and the acting commanding officer"
+	total_positions = 6
+	spawn_positions = 6
+	allow_additional = TRUE
+	scaled = TRUE
+	supervisors = "ARES and APOLLO"
 	selection_class = "job_working_joe"
 	flags_startup_parameters = ROLE_ADD_TO_DEFAULT|ROLE_WHITELISTED|ROLE_CUSTOM_SPAWN
 	flags_whitelist = WHITELIST_JOE
```
Comment (BeagleGaming1) [inline code/game/jobs/job/civilians/support/working_joe.dm:13]:
```suggestion
	flags_whitelist = WHITELIST_JOE
	gear_preset = /datum/equipment_preset/synth/working_joe
```

Comment (BeagleGaming1) [reply code/game/jobs/job/civilians/support/working_joe.dm:13]: You need a defualt gear preset otherwise it goes with default colonist human

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

File: code/modules/gear_presets/synths.dm [hunk 11]
```diff
@@ -486,11 +486,53 @@
...
+	new_human.equip_to_slot_or_del(new /obj/item/stack/sheet/metal/medium_stack(new_human.back), WEAR_IN_R_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/stack/sheet/glass/medium_stack(new_human.back), WEAR_IN_R_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/maintenance_jack(new_human), WEAR_J_STORE)
+
+
+/datum/equipment_preset/synth/working_joe/engi
+	name = "Synthetic - Hazmat Joe"
+
+/datum/equipment_preset/synth/working_joe/engi/load_gear(mob/living/carbon/human/new_human)
+	var/choice = rand(1,10)
```
Comment (BeagleGaming1) [inline code/modules/gear_presets/synths.dm:505]: image
Why have 1-10 if theres only 2.
Why have so many different if all is the same except for the uniform

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

File: code/modules/gear_presets/synths.dm [hunk 12]
```diff
@@ -486,10 +486,46 @@
...
+	new_human.equip_to_slot_or_del(new /obj/item/tool/extinguisher(new_human), WEAR_L_HAND)
 	new_human.equip_to_slot_or_del(new /obj/item/maintenance_jack(new_human), WEAR_J_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/storage/belt/utility/full(new_human), WEAR_WAIST)
+	new_human.equip_to_slot_or_del(new /obj/item/storage/pouch/tools/tank(new_human), WEAR_L_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/storage/pouch/construction(new_human), WEAR_R_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/device/lightreplacer(new_human.back), WEAR_IN_R_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/stack/sheet/metal/large_stack(new_human.back), WEAR_IN_R_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/stack/sheet/glass/large_stack(new_human.back), WEAR_IN_R_STORE)
+
+	switch(choice)
```
Comment (BeagleGaming1) [inline code/modules/gear_presets/synths.dm:524]: This needs to be before the other equips or it fails to equip it. 
image

File: code/modules/clothing/under/marine_uniform.dm [hunk 13]
```diff
@@ -842,12 +842,51 @@
...
+	desc = "A reinforced uniform used for Synthetic labor in hazardous areas. Comes with an additional layer for liquid hazards. Tomorrow, Together."
+	icon_state = "working_joe_overalls"
+	worn_state = "working_joe_overalls"
+	flags_inventory = CANTSTRIP
+	armor_melee = CLOTHING_ARMOR_LOW
+	armor_energy = CLOTHING_ARMOR_MEDIUMLOW
+	armor_bomb = CLOTHING_ARMOR_MEDIUMLOW
+	armor_bio = CLOTHING_ARMOR_MEDIUMHIGH
+	armor_rad = CLOTHING_ARMOR_HIGH
+	armor_internaldamage = CLOTHING_ARMOR_MEDIUMLOW
```
Comment (BeagleGaming1) [inline code/modules/clothing/under/marine_uniform.dm:890]: a lot of these vars are inherited and unneeded

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

File: code/game/machinery/rechargestation.dm [hunk 14]
```diff
@@ -157,21 +159,42 @@
...
+				to_chat(occupant, "Structural damage detected. Repairing...")
 				doing_stuff = TRUE
 				occupant.pain.recalculate_pain()
 			if(!doing_stuff && occupant.blood_volume < initial(occupant.blood_volume))
 				occupant.blood_volume = min(occupant.blood_volume + 10, initial(occupant.blood_volume))
-				to_chat(occupant, "Refreshing liquids...")
+				to_chat(occupant, "Fluid volume low. Refreshing liquids...")
 				doing_stuff = TRUE
+			if(!doing_stuff)
+				for(var/obj/limb/L in humanoid_occupant.limbs)
```
Comment (BeagleGaming1) [inline code/game/machinery/rechargestation.dm:176]: single letter var

File: code/game/machinery/rechargestation.dm [hunk 15]
```diff
@@ -157,21 +159,42 @@
...
+					if(L.implants.len)
+						doing_stuff = TRUE
+						to_chat(occupant, "Foreign material detected. Beginning removal process...")
+						for(var/obj/item/I in L.implants)
```
Comment (BeagleGaming1) [inline code/game/machinery/rechargestation.dm:180]: single letter var

File: code/game/machinery/rechargestation.dm [hunk 16]
```diff
@@ -157,21 +159,42 @@
...
+							if(!is_type_in_list(I,known_implants))
+								sleep(REMOVE_OBJECT_MAX_DURATION)
+								L.implants -= I
+								humanoid_occupant.embedded_items -= I
+								qdel(I)
+								to_chat(occupant, "Foreign object removed.")
+				for(var/datum/internal_organ/I in humanoid_occupant.internal_organs)
```
Comment (BeagleGaming1) [inline code/game/machinery/rechargestation.dm:187]: single letter var

File: code/game/machinery/rechargestation.dm [hunk 17]
```diff
@@ -157,21 +159,42 @@
...
+					if(I.robotic == ORGAN_ASSISTED||I.robotic == ORGAN_ROBOT) //this time the machine can *only* fix robotic organs
```
Comment (BeagleGaming1) [inline code/game/machinery/rechargestation.dm:188]:
```suggestion
					if(I.robotic == ORGAN_ASSISTED || I.robotic == ORGAN_ROBOT) //this time the machine can *only* fix robotic organs
```

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

**Pull request merged**